src/visualization/reducers/impression.ts

```diff
@@ -22,4 +22,38 @@ function sort(impression) {
         });
     }
     return impression;
+}
+
+function perceivedTime(impression){
+    
+        var startTime = 0;
+        var settleTime = 3000;
+        var mindWidth = 50;
+        var minHeight = 50;  
+        var index = 0;
+        var lastTime = startTime;
+        var settleTimeFound = "0";
+          
+        for(var evt of impression.events) {
+            if(evt.type === "Layout" && evt.state.tag === "IMG" && 
+            evt.state.layout.width >= mindWidth && evt.state.layout.height >= minHeight){
+                if((evt.time - lastTime) > settleTime){
+                    break;
+                }
+                lastTime = evt.time;
+            }else{
+                continue;
+            }
+        }
+
+        for (var evt of impression.events) {
+            if(evt.type === "Layout" && evt.state.tag === "IMG" && 
+                evt.state.layout.width >= mindWidth && evt.state.layout.height >= minHeight && evt.time <= lastTime ){
+                   
+                    evt["state"]["isPerceivedEvent"] = true;
+            } else{
+                evt["state"]["isPerceivedEvent"] = false;
+            }
```

Comment:
do we need to explicitly set this to false? If it isn't on the object at all we can trust it wasn't a perceivedperf event right?

Reply:
We can do that but I set it up to false to avoid undefined exceptions at any moment. Also the condition on change the opacity of the event is under the false value.  

---
In reply to: [177475290](https://github.com/Microsoft/clarity-tools/pull/41#discussion_r177475290) [](ancestors = 177475290)